ImageInLib/ImageInLib/src/imageInterpolation.c

```diff
@@ -3,257 +3,145 @@
 #include "imageInterpolation.h"
 #include<math.h>
 
-bool nearestNeighborInterpolation(dataType** originalImage, dataType** newImage, size_t imageLength, size_t imageWidth, size_t imageHeight, dataType originalSpacing, dataType newSpacing)
-{
-    if (originalImage == NULL || newImage == NULL)
-        return false;
+//3D images
 
-    size_t i, j, k, kn, x;
-    dataType k_int, k1, k2;
-
-    k_int = 0; kn = 0;
-
-    for (k = 0; k < imageHeight - 1; k++) {
-        k1 = originalSpacing * k;
-        k2 = originalSpacing * (k + 1);
-        do {
-            for (i = 0; i < imageLength; i++) {
-                for (j = 0; j < imageWidth; j++) {
-                    x = x_new(i, j, imageLength);
-                    if (k_int == 0) {
-                        newImage[kn][x] = originalImage[k][x];
-                    }
-                    else {
-                        if ((k_int - k1) < (k2 - k_int)) {
-                            newImage[kn][x] = originalImage[k][x];
-                        }
-                        else {
-                            newImage[kn][x] = originalImage[k + 1][x];
-                        }
-                    }
-                }
-            }
-            k_int = k_int + newSpacing;
-            kn = kn + 1;
-        } while (k_int < k2);
-    }
+Point3D getImageCoordFromRealCoord3D(Point3D image_point, Point3D image_origin, VoxelSpacing image_spacing, OrientationMatrix orientation) {
```

Comment:
In my opinion, this function does opposite transformation (from real world to image)